kura/org.eclipse.kura.cloud.mqtt.eclipseiot/src/main/java/org/eclipse/kura/cloud/mqtt/eclipseiot/internal/cloud/publisher/CloudPublisherImpl.java

```diff
@@ -0,0 +1,289 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ *******************************************************************************/
+package org.eclipse.kura.cloud.mqtt.eclipseiot.internal.cloud.publisher;
+
+import static java.util.Objects.nonNull;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.kura.KuraErrorCode;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloud.mqtt.eclipseiot.internal.cloud.CloudConnectionManagerImpl;
+import org.eclipse.kura.cloud.mqtt.eclipseiot.internal.cloud.CloudPublisherDeliveryListener;
+import org.eclipse.kura.cloud.mqtt.eclipseiot.internal.cloud.CloudServiceOptions;
+import org.eclipse.kura.cloudconnection.CloudConnectionManager;
+import org.eclipse.kura.cloudconnection.listener.CloudConnectionListener;
+import org.eclipse.kura.cloudconnection.listener.CloudDeliveryListener;
+import org.eclipse.kura.cloudconnection.message.KuraMessage;
+import org.eclipse.kura.cloudconnection.publisher.CloudPublisher;
+import org.eclipse.kura.configuration.ConfigurableComponent;
+import org.eclipse.kura.data.DataService;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.Filter;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CloudPublisherImpl
```

Comment:
It seems that the Eclipse IoT CloudPublisher does not publish the "apps certificate".
As a consequence, Kapua will not list the application identifiers.

Reply:
The apps certificate works with this refactoring only for "Cloudlets". With the new concept of Publisher and the concept of multiple applications bound to a single publisher it will have no sense.
We definitely need to update the basic publisher modifying the concept of appId that could be misleading.